rinja_parser/src/expr.rs

```diff
@@ -188,7 +190,56 @@ impl<'a> Expr<'a> {
     expr_prec_layer!(band, shifts, token_bitand);
     expr_prec_layer!(shifts, addsub, alt((tag(">>"), tag("<<"))));
     expr_prec_layer!(addsub, muldivmod, alt((tag("+"), tag("-"))));
-    expr_prec_layer!(muldivmod, filtered, alt((tag("*"), tag("/"), tag("%"))));
+    expr_prec_layer!(muldivmod, is_defined, alt((tag("*"), tag("/"), tag("%"))));
+
+    fn is_defined(i: &'a str, level: Level) -> ParseResult<'a, WithSpan<'a, Self>> {
+        let (_, level) = level.nest(i)?;
```

Comment:
I think we can to without the nesting check in here. We only traverse to a lower level, not the same or a higher level.

Reply:
Don't we need that for `Self::filtered(i, level)?` just below?

Reply:
No, you can just pass the `level` as is. In all other cases we also only increment the recursion level on recursions. The code cannot parse an expression like `y is defined is defined`, so no nesting happens in here.

Reply:
Good point! Removed this nesting.